src/SimpleSAML/XHTML/Template.php

```diff
@@ -206,16 +208,38 @@ public function asset(string $asset, ?string $module = null, bool $tag = true):
         }
 
         $file = new File($file);
+        $tag = $this->getAssetTagForFile($file);
 
-        $tag = $this->configuration->getVersion();
-        if ($tag === 'master') {
-            $tag = strval($file->getMtime());
-        }
-        $tag = substr(hash('md5', $tag), 0, 5);
+        // Use the `assets.secretsalt` to enhance security.
+        // Do not make it easy to guess the underlying SSP version.
+        $salt = $this->configuration->getOptionalStringNested(
+            'assets.secretsalt',
+            'assets.secretsalt.default',
+        );
+        $mac = hash_hmac('sha256', $tag, $salt, true);
+        $tag = substr(base64_encode($mac), 0, $tagLength);
```

Comment:
From some oidc work I learned  that base64 encodings are not url safe since they can produce `+`, `/` and `=`.  Some languages have a b64 url safe version but I did not see one for php

Reply:
this is easily fixed with this: